l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -459,7 +482,14 @@ contract ExecutorFacet is Base, IExecutor {
                 l2ToL1LogsHash,
                 _stateDiffHash,
                 _batch.bootloaderHeapInitialContentsHash,
-                _batch.eventsQueueStateHash
+                _batch.eventsQueueStateHash,
+                // for each blob we have: 
+                // linear hash (hash of preimage from system logs) and 
+                // output hash of blob commitments: keccak(versioned hash || opening point || evaluation value)
+                _blob1Hash,
+                _blobCommitments[0],
+                _blob2Hash,
+                _blobCommitments[1]
```

Comment:
Both of these values can be zero. On L2, an operator may have data that fits into one blob and so `blob2Hash == 0`. On L1, an operator may provide commitment to only one of the blob, then `_blobCommitments[1]`.

What is implications of (`blob2Hash == 0` && `_blobCommitments[1] != 0`) || (`blob2Hash != 0` && `_blobCommitments[1] == 0`)?

Reply:
its a good call, ideally both would be zero if only one blob was being used. but in the event that only one of the hash, commitment are zero then we would likely have the batch be committed but proving should fail. I'm all for adding a require to catch it at the commit step

Reply:
Here I added one more comment on that topic: https://github.com/matter-labs/era-contracts/pull/138#discussion_r1450153684. The option of zero is generally not correct